lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -146,6 +146,144 @@ def formatted_message_and_backtrace(colorizer, indentation)
+
+            options = options.merge(
+              :failure_lines          => [],
+              :extra_detail_formatter => sub_failure_list_formatter(exception, options[:message_color]),
+              :detail_formatter       => multiple_failure_sumarizer(exception,
+                                                                    options[:detail_formatter],
+                                                                    options[:message_color])
+            )
+
+            options[:description_formatter] &&= Proc.new {}
+
+            return options unless exception.aggregation_metadata[:from_around_hook]
+            options[:backtrace_formatter] = EmptyBacktraceFormatter
+            options
+          end
+
+          def multiple_exceptions_not_met_error?(exception)
+            return false unless defined?(RSpec::Expectations::MultipleExpectationsNotMetError)
+            RSpec::Expectations::MultipleExpectationsNotMetError === exception
+          end
+
+          def multiple_failure_sumarizer(exception, prior_detail_formatter, color)
+            lambda do |example, colorizer, indentation|
+              summary = if exception.aggregation_metadata[:from_around_hook]
+                          "Got #{exception.exception_count_description}:"
+                        else
+                          "#{exception.summary}."
+                        end
+
+              summary = colorizer.wrap(summary, color || RSpec.configuration.failure_color)
+              return summary unless prior_detail_formatter
+              "#{prior_detail_formatter.call(example, colorizer, indentation)}\n#{indentation}#{summary}"
+            end
+          end
+
+          def sub_failure_list_formatter(exception, message_color)
+            common_backtrace_truncater = CommonBacktraceTruncater.new(exception)
+
+            lambda do |failure_number, colorizer, indentation|
+              exception.all_exceptions.each_with_index.map do |failure, index|
+                options = with_multiple_error_options_as_needed(
+                  failure,
+                  :description_formatter   => :failure_slash_error_line.to_proc,
+                  :indentation             => indentation.length,
+                  :message_color           => message_color || RSpec.configuration.failure_color,
+                  :skip_shared_group_trace => true
+                )
+
+                failure   = common_backtrace_truncater.with_truncated_backtrace(failure)
+                presenter = ExceptionPresenter.new(failure, @example, options)
+                presenter.fully_formatted("#{failure_number}.#{index + 1}", colorizer)
+              end.join
+            end
+          end
+
+          # @private
+          # Used to prevent a confusing backtrace from showing up from the `aggregate_failures`
+          # block declared for `:aggregate_failures` metadata.
+          module EmptyBacktraceFormatter
```

Comment:
It's meant to be an empty version of https://github.com/rspec/rspec-core/blob/master/lib/rspec/core/backtrace_formatter.rb, which is called `BacktraceFormatter`, hence `EmptyBacktraceFormatter`.  I don't feel strongly about it.  You mind if we go with this for now (particularly since it's already called this in master -- it just moved files in this PR) and after this PR is merged if you want to rename it and `BacktraceFormatter` to something else you can?